### ISSUE TITLE
Improve Jacobian debug script

### ISSUE THREAD

**Issue by dschwen** [idaholab/moose#4182]
Due to the HPC outage some stuff is piling up. This will be the next generation of the Jacobian debugger that utilizes the DOFMap output. This issue depends on PRs #4147 and #4175 to be merged first.

Meta: This is a bit of a painful situation. I'm developing this on a branch in which I merged the local branches corresponding to the two PRs above. There is a lot of cherry picking going on whenever I update those branches, and I expect I'll have to cherry-pick the changes from this new branch into a clean feature branch before doing a PR. Or is there a better more gitolicious way to work?

**Comment by dschwen** [comment]
I'll open a new issue when I continue working on this.
